### ISSUE TITLE
pkgs.pkgsCross.raspberryPi.stdenv works but not pkgs.pkgsCross.raspberryPi.clangStdenv

### ISSUE THREAD

**Issue by tfc** [NixOS/nixpkgs#61205]
## Issue description

I tried to crosscompile a simple tiny C++ application on x86_64 for the Raspberry Pi using `pkgs.pkgsCross.raspberryPi`.
This works nicely with `stdenv` from that set, but not with `clangStdenv`.

### Steps to reproduce

The code is here: https://github.com/tfc/nix_cpp_crosscompile/

Run the following nix command:

```bash
$  nix-build release.nix -A raspberry-clang
```

This will fail while trying to build `/nix/store/2kc1pn93i94k4c6cjgpsy3bssizi8760-cross-compile-example-armv6l-unknown-linux-gnueabihf.drv`.
I am not sure how to debug this as in the cross compilation setup expressions i quickly feel lost.

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

```bash
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.0.9, NixOS, 19.03.172500.aade6ded796 (Koi)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.2.2`
 - channels(tfc): `"home-manager-19.03"`
 - channels(root): `"nixos-19.03.172500.aade6ded796, nixos-unstable-19.09pre178539.d457e31d9ec"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`
```

**Comment by matthewbauer** [comment]
Yeah this is due to the awkwardness of how stdenv works. It is the one common dependency of everything in Nixpkgs. It's this weird special case that connects the cross stages. Replacing it with clangStdenv doesn't work directly. Instead, you can do:

```
pkgs.pkgsCross.raspberryPi.buildPackages.clangStdenv
```

**Comment by marius851000** [comment]
I figured the error is that stdenv.hostPlatform.useLLVM is undefined, but it should be true. changing `useLLVM` = stdenv.hostPlatform.useLLVM or false;` to `useLLVM = stdenv.hostPlatform.useLLVM or true;` in `pkgs/development/compilers/llvm/7/compiler-rt.nix` allow this project to compile. I will take a look at why it is undefined.
